x-pack/test/apm_api_integration/basic/tests/transaction_groups/error_rate.ts

```diff
@@ -36,11 +35,78 @@ export default function ApiTest({ getService }: FtrProviderContext) {
 
       it('returns the transaction error rate', async () => {
         const response = await supertest.get(
-          `/api/apm/services/opbeans-node/transaction_groups/error_rate?start=${start}&end=${end}&uiFilters=${uiFilters}`
+          `/api/apm/services/opbeans-java/transaction_groups/error_rate?start=${start}&end=${end}&uiFilters=${uiFilters}`
         );
 
         expect(response.status).to.be(200);
-        expect(response.body).to.eql(expectedErrorRate);
+
+        expect(response.body).to.eql({
+          noHits: false,
+          erroneousTransactionsRate: [
```

Comment:
IMHO this is not really helpful. Can we avoid a snapshot test? e.g. we can count the buckets, or compare the avg only, etc.

Reply:
I agree, we should spent a little more time finding the right things to test instead of blindly testing the full output (I've been very guilty of this myself). Testing the entire output like this makes the test prone to false positives, where it fails when new properties are added (which is not in itself an issue).

The challenge by NOT doing snapshot testing is that we miss stuff and bugs can slip in without us noticing.
I tried to think of the important things to verify, while still making it flexible enough to allow a developer to change the output (eg. add new properties).

```ts
const { erroneousTransactionsRate } = response.body;

// we want to assert that the range is correct
it("has the correct start date", async () => {
  expect(erroneousTransactionsRate[0].x).to.be("...");
});

it("has the correct end date", async () => {
  expect(_.last(erroneousTransactionsRate).x).to.be("...");
});

it("has the correct number of buckets", async () => {
  expect(erroneousTransactionsRate.length).to.be(1337);
});

// we want to assert that the timeseries values align with the average calculated in elasticsearch (not sure they are identical so might not be possible but worth a shot)
it("has the correct calculation for average", async () => {
  const average = _.meanBy(erroneousTransactionsRate, (p) => p.y);
  expect(average).to.be(response.body.average);
});

```

Reply:
I like what you suggested @sqren, but I feel that in the test scenarios you described above we are not testing what really matters for the API, that is the error rate.  The average is not calculated in ES, it also uses lodash mean, so even if we change the way we calculate the error rate these tests will never fail. 

I agree that we should avoid snapshot tests, but in the end, we are testing the return of an API and we must find a balance between these two approaches.